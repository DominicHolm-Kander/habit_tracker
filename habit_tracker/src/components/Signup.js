import React, { Component } from 'react'

export class Signup extends Component {

    handleChangeOfUsername = (event) => {
        this.setState({username: event.target.value})
    }
    handleChangeOfPassword = (event) => {
        this.setState({password: event.target.value})
    }
    handleChangeOfFn = (event) => {
        this.setState({fn: event.target.value})
    }
    handleChangeOfLn = (event) => {
        this.setState({ln: event.target.value})
    }
    

    render() {
        return (
            <div>
                <h3>Or if you're a first-time visitor, sign up below!</h3>
                <form>
                    <label for="firstName">First Name
                    <input type="text" className = "firstName" name="fn"onChange={(event) => this.handleChangeOfFn(event)}></input>
                    </label>
                    <label for="lastName">Last Name
                    <input type="text" className = "lastName" name="ln"onChange={(event) => this.handleChangeOfLn(event)}></input>
                    </label>
                    <label for="userName">Username
                    <input type="text" className = "userName" name="userName" onChange={(event) => this.handleChangeOfUsername(event)}></input>
                    </label>
                    <label for="password">Password
                    <input type="text" className = "password" name="password"onChange={(event) => this.handleChangeOfPassword(event)}></input>
                    </label>
                    <button type="button" id="signUpButton" onClick={() => this.props.signup(this.state)}>Sign up</button>
                </form>
            </div>
        )
    }
}

export default Signup
