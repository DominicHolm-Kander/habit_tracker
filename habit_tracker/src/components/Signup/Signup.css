.btn{
    margin-top:40px;
    font-family: 'Bowlby One SC', cursive;
}
.col-md-4, .mb-3{
    margin-top:500px;
}
h4{
    margin-left:20px;
    margin-top:40px;
    font-family:'Bangers', cursive;
}
body{
    background-color: greenyellow;
}
/* .signup{
    margin-top:80px;
} */