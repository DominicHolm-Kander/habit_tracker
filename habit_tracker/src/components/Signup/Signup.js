import React, { Component } from 'react'
import './Signup.css'
export class Signup extends Component {

    handleChangeOfUsername = (event) => {
        this.setState({username: event.target.value})
    }
    handleChangeOfPassword = (event) => {
        this.setState({password: event.target.value})
    }
    handleChangeOfFn = (event) => {
        this.setState({fn: event.target.value})
    }
    handleChangeOfLn = (event) => {
        this.setState({ln: event.target.value})
    }
    

    render() {
        return (
          <div>
            <h4>If you're a first-time visitor, sign up below!</h4>
    <div className="signup">
            <form clasName="needs-validation" novalidate>
              <div clasName="form-row">
                <div clasName="col-md-4 mb-3">
                  {/* <label for="validationCustom01">First name</label> */}
                  <input type="text" clasName="form-control" id="validationCustom01" placeholder="First name" name="fn" required onChange={(event) => this.handleChangeOfFn(event)}></input>
                  
                </div>
              </div>
              <div clasName="col-md-4 mb-3">
                {/* <label for="validationCustom02">Last name</label> */}
                <input type="text" clasName="form-control" id="validationCustom02" placeholder="Last name" name="ln" onChange={(event) => this.handleChangeOfLn(event)} required></input>
                </div>
              
              <div clasName="col-md-4 mb-3">
                {/* <label for="validationCustom02">Username</label> */}
                <input type="text" clasName="form-control" id="validationCustom02" placeholder="Username" name="userName" onChange={(event) => this.handleChangeOfUsername(event)} required></input>
                </div>
              <div clasName="col-md-4 mb-3">
                {/* <label for="validationCustom02">Password</label> */}
                <input type="password" clasName="form-control" id="validationCustom03" placeholder="Password" name="password" onChange={(event) => this.handleChangeOfPassword(event)} required></input>
                </div>
              
             
              <div clasName="form-row">
                <div clasName="col-md-6 mb-3">
                  <button className="btn btn-primary" type="submit" onClick={() => this.props.signup(this.state)}>Sign up</button>
                </div>
              </div>
            </form>
            </div>
          </div>
        )
    }
}

export default Signup
