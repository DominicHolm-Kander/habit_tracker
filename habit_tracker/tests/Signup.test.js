import React from 'react'
import { shallow } from 'enzyme'
import Signup from "../src/components/Signup"

describe("shallow Login", () => {
    let wrapper;

    beforeEach(() => (wrapper = shallow(<Signup />)))
    it("should render correctly", () => {
        expect(wrapper).toMatchSnapshot();
    })
    it("should have 4 input fields and a button", () => {
        expect(wrapper.find("input").length).toEqual(4);
        expect(wrapper.find("button").length).toEqual(1);
    })
})