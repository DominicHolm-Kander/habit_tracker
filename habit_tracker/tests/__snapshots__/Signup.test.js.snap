// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow Login should render correctly 1`] = `
<div>
  <h3>
    Or if you're a first-time visitor, sign up below!
  </h3>
  <form>
    <label
      for="firstName"
    >
      First Name
      <input
        className="firstName"
        name="fn"
        onChange={[Function]}
        type="text"
      />
    </label>
    <label
      for="lastName"
    >
      Last Name
      <input
        className="lastName"
        name="ln"
        onChange={[Function]}
        type="text"
      />
    </label>
    <label
      for="userName"
    >
      Username
      <input
        className="userName"
        name="userName"
        onChange={[Function]}
        type="text"
      />
    </label>
    <label
      for="password"
    >
      Password
      <input
        className="password"
        name="password"
        onChange={[Function]}
        type="text"
      />
    </label>
    <button
      id="signUpButton"
      onClick={[Function]}
      type="button"
    >
      Sign up
    </button>
  </form>
</div>
`;
